app/queries/keywords_query.rb

```diff
@@ -9,14 +9,28 @@ def initialize(scope, filter_params)
   end
 
   def call
-    return scope unless defined?(filter_params[:adwords_url_contains]) && filter_params[:adwords_url_contains]
-
-    filter_ads_top_urls(filter_params[:adwords_url_contains])
+    if filter_params[:adwords_url_contains]
+      filter_ads_top_urls(filter_params[:adwords_url_contains])
+    elsif filter_params[:word] && filter_params[:match_at_least]
+      filter_result_urls(filter_params[:word], filter_params[:match_at_least])
+    else
+      scope
+    end
+  rescue NoMethodError
+    scope
   end
 
   private
 
   def filter_ads_top_urls(word)
     scope.where("array_to_string(ads_top_urls, '||') ILIKE ?", "%#{word}%")
   end
+
+  def filter_result_urls(word, number_of_matches)
```

Comment:
```suggestion
  def filter_ads_top_urls
    word = filter_params[:adwords_url_contains]&.downcase
    
    return unless word
    
    scope.where("array_to_string(ads_top_urls, '||') ILIKE ?", "%#{word}%")
  end

  def filter_result_urls
    # same here, include the condition logic and the filter param processing
```

Let's leverage OOP to make the conditions and calls above less complex.

Then the call method becomes much easier to read: it just calls each filtering method in the expected order. The logic for each filtering is then contained inside 1 method (and not spread between `call` and the filtering method).